tests/integration/server/proposals_api_test.py

```diff
@@ -13,7 +13,115 @@
 # limitations under the License.
 # -----------------------------------------------------------------------------
 """Integration tests for proposals APIs"""
+import requests
 
-# TODO: Add tests for the proposals API
-# TODO: Add a test to check that users from the manager chain are in the
-#      approvers list.
+from tests.utilities import (
+    add_role_member,
+    create_test_user,
+    create_test_role,
+    log_in,
+    update_proposal,
+)
+
+TEST_USERS = [
+    {
+        "name": "Ash K",
+        "username": "ashketcham1",
+        "password": "password321",
+        "email": "gotta.catchem@all.com",
+    },
+    {
+        "name": "Garen G",
+        "username": "gareng123",
+        "password": "password123",
+        "email": "gareng@testuser.com",
+    },
+]
+
+TEST_ROLES = [
+    {"name": "NEXT Maintainer", "owners": "", "administrators": ""},
+    {"name": "Teacher", "owners": "", "administrators": ""},
+]
+
+
+def setup_module():
+    """Setting up users and roles for various tests """
+    with requests.Session() as session:
+        # Creating test users
+        for user in TEST_USERS:
+            user_id = create_test_user(session, user).json()["data"]["user"]["id"]
+            user["id"] = user_id
+
+        # Creating test roles
+        for role in TEST_ROLES:
+            role["owners"] = TEST_USERS[0]["id"]
+            role["administrators"] = TEST_USERS[0]["id"]
+            role_id = create_test_role(session, role).json()["data"]["id"]
+            role["id"] = role_id
+
+
+def test_proposal_api_as_owner():
+    """Test updating a proposal as role owner"""
+    with requests.Session() as session:
+
+        # Logging in as role owner
+        credentials_payload = {
+            "id": TEST_USERS[0]["username"],
+            "password": TEST_USERS[0]["password"],
+        }
+        auth_response = log_in(session, credentials_payload)
+
+        # Creating Proposal
+        proposal_payload = {"id": TEST_USERS[1]["id"]}
+        proposal_id = add_role_member(
+            session, TEST_ROLES[0]["id"], proposal_payload
+        ).json()["proposal_id"]
+        # proposal_id = proposal_res.json()["proposal_id"]
+
+        # Updating proposal created above as role owner
+        update_proposal_payload = {
+            "status": "APPROVED",
+            "reason": "Need on team.",
+            "metadata": "approval-date:2019-05-09",
```

Comment:
Update proposal API endpoint does not take a metadata field, only status and reason.